Models/PhotosynthesisModel.cs

```
@@ -0,0 +1,229 @@
+        {
+            Temperature.UpdateAirTemperature(time);
+            Radiation.UpdateRadiationValues(time);
+            var sunAngle = Solar.SunAngle(time);            
+            Canopy.CalcCanopyStructure(sunAngle);
+
+            return IsSensible();
+        }
+
+        /// <summary>
+        /// Check if the basic conditions for photosynthesis to occur are met
+        /// </summary>
+        private bool IsSensible()
+        {
+            var CPath = Canopy.Canopy;
+            var temp = Temperature.AirTemperature;
+
+            bool[] tempConditions = new bool[4]
+            {
+                temp > CPath.Pathway.ElectronTransportRateParams.TMax,
+                temp < CPath.Pathway.ElectronTransportRateParams.TMin,
+                temp > CPath.Pathway.MesophyllCO2ConductanceParams.TMax,
+                temp < CPath.Pathway.MesophyllCO2ConductanceParams.TMin
+            };
+
+            bool invalidTemp = tempConditions.Any(b => b == true);
+            bool invalidRadn = Radiation.Total <= double.Epsilon;
+
+            if (invalidTemp || invalidRadn)
+                return false;
+            else
+                return true;
+        }
+
+        /// <summary>
+        /// Determine the total potential biomass for the day under ideal conditions
+        /// </summary>
+        public void CalculatePotential(out double intercepted, out double[] assimilations, out double[] sunlitDemand, out double[] shadedDemand)
+        {
+            // Water demands
+            intercepted = 0.0;
+            sunlitDemand = new double[iterations];
+            shadedDemand = new double[iterations];
+            assimilations = new double[iterations];
+
+            for (int i = 0; i < iterations; i++)
+            {
+                double time = start + i * timestep;
+
+                // Note: double arrays default value is 0.0, which is the intended case if initialisation fails
+                if (!TryInitiliase(time)) continue;
+
```

Is it valid for the model to continue if TryInitialise fails?
If not, then it should raise an exception.

---

Yes, TryInitialise occurs each timestep of the calculation. If it fails, it just means that there is no need to calculate photosynthesis for that time, which occurs when there is insufficient radiation or extreme temperature. In such a case, it is still possible that the other timesteps have valid conditions for photosynthesis, so the model needs to continue.